tokio-util/src/io/sync_bridge.rs

```diff
@@ -153,4 +153,18 @@ impl<T: Unpin> SyncIoBridge<T> {
     pub fn into_inner(self) -> T {
         self.src
     }
+
+    /// Returns a shared reference to the inner resource.
+    ///
+    /// It is inadvisable to directly read/write from the underlying resource.
+    pub fn get_ref(&self) -> &T {
+        &self.src
+    }
+
+    /// Returns a mutable reference to the inner resource.
+    ///
+    /// It is inadvisable to directly read/write from the underlying resource.
+    pub fn get_mut(&mut self) -> &mut T {
+        &mut self.src
+    }
```

Comment:
We probably should use the AsRef/AsMut traits instead? I think we do that in most places where we provide accessor to the inner value.

Reply:
There's precedent for both (`Framed` for example offers explicit `get_{ref,mut}` methods, whereas `stream_wrappers` implement `{AsMut,AsRef}`.

I now switched to implement the traits instead here.